pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -204,6 +204,23 @@ public void build() throws Exception {
     }
   }
 
+  private boolean containsInvalidInvertedIndex() {
+    int numValues = segmentIndexCreationInfo.getTotalDocs();
+    for (String columnName : config.getInvertedIndexCreationColumns()) {
+      LOGGER.info("Evaluating inverted index size for column {} ", columnName);
+      long numBytes = numValues * V1Constants.Numbers.INTEGER_SIZE;
+      ColumnIndexCreationInfo indexCreationInfo = indexCreationInfoMap.get(columnName);
+      int cardinality = indexCreationInfo.getDistinctValueCount();
+      LOGGER.info("Cardinality of column {} is {} ", columnName, String.valueOf(cardinality));
+      numBytes += cardinality * indexCreationInfoMap.get(columnName).getLegnthOfLongestEntry();
+      LOGGER.info("Estimated inverted index size for column {} is {}, should be < 2GB", columnName, numBytes);
+      if (numBytes >= 2147483648L) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   private void buildStarTree() throws Exception {
```

Comment:
why is this part of buildStarTree method. Its better to call validate method in seal method and keep all the validation in one place.